### DISCUSSION TITLE
Filebeat modules and timezone

### DISCUSSION THREAD

**Discussion by Sahale** [Security-Onion-Solutions/securityonion#8445] [Unsupported Versions] [answered]
Hello!

SO 2.3.130.
We use Fortinet and PaloAlto filebeat modules to process events. Both Forti and PA send their events with non-UTC time (i.e. +01:00). And SO parse it with +01:00 from correct time. So to see new events I need to select some time in future.
Fortinet module has var.tz_offset option, but it doesn't fix this problem.

I created logstash pipelines (they look for event.module value) and elastic ingest pipelines to set correct timestamp and it seems to be working, but not for all events. For example, some FortiMail events get correct timestamp, but some don't. I can't find why it's working in such strange way.

Maybe there is much easier way to fix that? Thank you!

Part of logstash pipeline:
```
...
output {
  if [event][module] == "fortinet" {
    elasticsearch {
      pipeline => "foo-fortinet"
      hosts => "{{ ES }}"
{% if salt['pillar.get']('elasticsearch:auth:enabled') is sameas true %}
      user => "{{ ES_USER }}"
      password => "{{ ES_PASS }}"
{% endif %}
      index => "so-fortinet"
      ssl => true
      ssl_certificate_verification => false
    }
  }
}
```

Elastic pipeline:

```
{
"description": "Pipeline to correct Fortinet timestamp",
"processors": [
    { "date":        { "field": "rsa.time.event_time", "timezone": "+0100", "formats": [ "ISO8601"] ,  "on_failure": [ ... ] } },
    { "set":         { "field": "testfield", "value": "To_Check_If_Its_Working", "ignore_failure": true } }
],
"version": 2
}
```

**Comment by dougburks** [comment]
> Fortinet module has var.tz_offset option, but it doesn't fix this problem.

Seems like that option is designed to solve the issue that you're having, so you might double-check your configuration.

Other options you might want to consider:
- configure the pipeline to use the ingest time instead
- change the timezone on the original device to UTC

**Comment by Sahale** [comment, marked as answer]
Well, something strange happened. After upgrade to .140 FortiMail tz_offset became fixed somehow. Didn't change anything, so can't say what's really helped.

PaloAlto doesn't have tz_offset option. But while I was playing with filebeat/logstash/PaloAlto parameters, I changed "Log format" on PaloAlto from BSD to IETF and this solved the problem.
